### DISCUSSION TITLE
How to pass options from Nuxt module to Nitro plugin ?

### DISCUSSION THREAD

**Discussion by pand-app** [nuxt/nuxt#24020] [Questions] [answered]
Hello,

How to pass options of a module to nitro plugin ? I cannot use useRuntime config in nitro plugin.
I can import it from `#imports` in playground but it's not working in a "real" project. 
My current workaround is to pass options to process.env the cast back in my nitro plugin 🤦‍♂️

Thanks !

**Comment by xjccc** [comment]
[addServerPlugin](https://nuxt.com/docs/api/kit/nitro#addserverplugin)

**Reply by pand-app** [reply]
addServerPlugin don't have any parameter to pass options.

**Reply by xjccc** [reply, marked as answer]
like this?

[nuxt-module/color-mode](https://github.com/nuxt-modules/color-mode/blob/master/src/module.ts#L39)

[server-plugin](https://github.com/nuxt-modules/color-mode/blob/master/src/runtime/plugin.server.ts#L5)

**Reply by pand-app** [reply]
This is not as beautiful than expected, but it's working, only if I set `write: true`
=> https://gitlab.com/qonfucius/nuxt-prometheus-module/-/merge_requests/3

Thanks

**Comment by pawelgur** [comment]
While answer above works, though worth mentioning alternative solution for future reference.

Not sure why @Nainterceptor can't use runtime config, because this seem to work too and requires less boilerplate:
```
// module
export default defineNuxtModule({
  setup (options, nuxt) {
    const { resolve } = createResolver(import.meta.url)

    nuxt.options.runtimeConfig.pluginOption = 123;
    addServerPlugin(resolve('./nitro-plugin.ts'));
  }
```
```
// nitro plugin
export default defineNitroPlugin((nitroApp) => {
     console.log('testOption', useRuntimeConfig().pluginOption);
})
```

**Reply by TheAlexLichter** [reply]
This is the way 👍

**Reply by xjccc** [reply]
thank you for show the best way🫡
